velox/functions/prestosql/tests/BinaryFunctionsTest.cpp

```diff
@@ -697,4 +697,69 @@ TEST_F(BinaryFunctionsTest, toIEEE754Bits32) {
       hexToDec("FF7FFFFF"),
       toIEEE754Bits32(std::numeric_limits<float>::lowest()));
 }
+
+TEST_F(BinaryFunctionsTest, rpad) {
+  const auto rpad = [&](std::optional<StringView> arg,
+                        std::optional<StringView> key,
+                        std::optional<int32_t> size) {
+    return evaluateOnce<std::string>("rpad(c0, c1, c2)", makeRowVector({
+    vectorMaker_.flatVectorNullable<StringView>(
+      {arg}, VARBINARY()),
+      vectorMaker_.flatVectorNullable<StringView>(
+      {key}, VARBINARY()),
+    vectorMaker_.flatVectorNullable<int32_t>(
+      {size}, INTEGER())
+    })); 
+  };
+  EXPECT_EQ(rpad(StringView("1234"),StringView("45"), 14), ("12344545454545"));
```

Comment:
I think all the StringView calls for the parameters would be redundant. Using the underlying std::strings to pass to rpad lambda should be fine. Can you try that ? Same comment for lpad.

Reply:
i think the issue with using the underlying std:string is that it gets mapped to a varchar by CppToType(). due to this, it would want to go into the varchar lpad and rpad functions. This way, they get mapped to a varbinary value

Reply:
So using VARBINARY() for the vectors in evaluateOnce should be sufficient to trigger the varbinary functions I think.

StringView is a class for a view on string. Using StringView in the lpad and rpad lambda as function parameters is sufficient.

std::string in the calls to lpad and rpad should implicitly convert to StringView. Is that not the case ?

Reply:
worked with aditi offline and resolved this issue to make the conversions from string to varbinaries properly